tests/integration/build_test.sh

```diff
@@ -0,0 +1,183 @@
+  fi
+}
+
+function test_kernel_build_gcc_x86_64_no_env()
+{
+  local kw_build_cmd
+  local build_status
+  local build_result
+  local build_status_log
+
+  kw_build_cmd='kw build'
+  container_exec "$CONTAINER" "cd ${KERNEL_TREE_PATH_CONTAINER} && ${kw_build_cmd} > /dev/null 2>&1"
+  assert_equals_helper "kw build failed for ${CONTAINER}" "(${LINENO})" 0 "$?"
+
+  # Verify kernel binary exists
+  kernel_binary_path=$(container_exec "$CONTAINER" "find ${KERNEL_TREE_PATH_CONTAINER}/arch/x86/boot/ -type f -name 'bzImage'")
+  kernel_binary_found=$([[ -n "$kernel_binary_path" ]] && echo 0 || echo 1)
+  assert_equals_helper "Kernel binary not found for ${CONTAINER}" "$LINENO" "0" "$kernel_binary_found"
+
+  # Retrieve the build status log from the database
+  build_status_log=$(container_exec "$CONTAINER" "sqlite3 ~/.local/share/kw/kw.db \"SELECT * FROM statistics_report\" | tail --lines=1")
+
+  # Extract the build status and result from the log
+  build_status=$(printf '%s' "$build_status_log" | cut --delimiter='|' --fields=2)
+  assert_equals_helper "Build status check failed for ${CONTAINER}" "$LINENO" 'build' "$build_status"
+
+  build_result=$(printf '%s' "$build_status_log" | cut --delimiter='|' --fields=3)
+  assert_equals_helper "Build result check failed for ${CONTAINER}" "$LINENO" 'success' "$build_result"
+}
+
+function test_kernel_build_cpu_scaling_option()
+{
+  local build_status
+  local build_result
+  local build_status_log
+  local cpu_scaling_percentage=50
+
+  # Execute the test script inside the container
+  # The test script will run the kw build command with the --cpu-scaling 50 option
+  container_exec "$CONTAINER" "cd ${KERNEL_TREE_PATH_CONTAINER} && kw_build_cpu_scaling_monitor ${cpu_scaling_percentage} > /dev/null 2>&1"
+  assert_equals_helper "kw build --cpu-scaling 50 failed for ${CONTAINER}" "(${LINENO})" 0 "$?"
+
+  # For more details about this test, check the file:
+  # tests/integration/podman/scripts/kw_build_cpu_scaling_monitor
+}
+
+function test_kernel_build_llvm()
+{
+  local kw_build_cmd
+  local build_status
+  local build_result
+  local build_status_log
+
+  kw_build_cmd='kw build --llvm'
+  container_exec "$CONTAINER" "cd ${KERNEL_TREE_PATH_CONTAINER} && ${kw_build_cmd} > /dev/null 2>&1"
+  assert_equals_helper "kw build failed for ${CONTAINER}" "(${LINENO})" 0 "$?"
+
+  # Retrieve the build status log from the database
+  build_status_log=$(container_exec "$CONTAINER" "sqlite3 ~/.local/share/kw/kw.db \"SELECT * FROM statistics_report\" | tail --lines=1")
+
+  # Extract the build status and result from the log
+  build_status=$(printf '%s' "$build_status_log" | cut --delimiter='|' --fields=2)
+  assert_equals_helper "Build status check failed for ${CONTAINER}" "$LINENO" 'build' "$build_status"
+
+  build_result=$(printf '%s' "$build_status_log" | cut --delimiter='|' --fields=3)
+  assert_equals_helper "Build result check failed for ${CONTAINER}" "$LINENO" 'success' "$build_result"
```

Comment:
I considered modularizing this part, but I realized it would bring back the issue where `assert_equals_helper` always points to the same line when an error occurs, making debugging significantly more challenging.